src/components/PeopleFilters/PeopleFilters.tsx

```diff
@@ -0,0 +1,157 @@
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+
+    const params = new URLSearchParams(searchParams);
+    const currentCenturies = params.getAll('centuries');
+
+    if (currentCenturies.includes(century)) {
+      const newCenturies = currentCenturies.filter(
+        currCentury => currCentury !== century,
+      );
+
+      params.delete('centuries');
+      newCenturies.forEach(currCentury =>
+        params.append('centuries', currCentury),
+      );
+    } else {
+      params.append('centuries', century);
+    }
+
+    setSearchParams(params);
+  };
+
+  const toggleAllCenturies = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    const params = new URLSearchParams(searchParams);
+
+    params.delete('centuries');
+    setSearchParams(params);
+  };
+
+  const toggleGender = (
+    gender: 'all' | 'm' | 'f',
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    const params = new URLSearchParams(searchParams);
+
+    if (gender === 'all') {
+      params.delete('sex');
+    } else {
+      params.set('sex', gender);
+    }
+
+    setSearchParams(params);
+  };
+
+  return (
+    <nav className="panel">
+      <p className="panel-heading">Filters</p>
+
+      <p className="panel-tabs" data-cy="SexFilter">
+        <a
+          className={cn({ 'is-active': !sex })}
+          href="/people"
+          onClick={event => toggleGender('all', event)}
+        >
+          All
+        </a>
+        <a
+          className={cn({ 'is-active': sex === 'm' })}
+          href="/people?sex=m"
+          onClick={event => toggleGender('m', event)}
+        >
+          Male
+        </a>
+        <a
+          className={cn({ 'is-active': sex === 'f' })}
+          href="/people?sex=f"
+          onClick={event => toggleGender('f', event)}
+        >
+          Female
+        </a>
+      </p>
+
+      <div className="panel-block">
+        <p className="control has-icons-left">
+          <input
+            data-cy="NameFilter"
+            type="search"
+            className="input"
+            placeholder="Search"
+            value={query}
+            onChange={event => {
+              handleQuery(event);
+            }}
+          />
+
+          <span className="icon is-left">
+            <i className="fas fa-search" aria-hidden="true" />
+          </span>
+        </p>
+      </div>
+
+      <div className="panel-block">
+        <div className="level is-flex-grow-1 is-mobile" data-cy="CenturyFilter">
+          <div className="level-left">
+            {centuriesList.map(century => (
+              <a
+                data-cy="century"
+                className={cn('button mr-1', {
+                  'is-info': centuries.includes(century.toString()),
+                })}
+                key={century}
+                href={`/people?centuries=${century}`}
```

Comment:
The `href` attribute here is set to `/people?centuries=${century}`, which might not reflect the current state of other filters. Consider using the `searchParams` to construct the URL dynamically to preserve other query parameters.